src/Service/Bulk/BulkCreateUsersService.php

```diff
@@ -0,0 +1,294 @@
+
+class BulkCreateUsersService
+{
+    private array $lineItemIds = [];
+    private array $lineItemSlugs = [];
+
+    private int $userGroupBatchCount = 0;
+    private int $groupIndex = 0;
+
+    private string $generatedUsersFilePath;
+
+    private LineItemRepository $lineItemRepository;
+    private AssignmentRepository $assignmentRepository;
+    private Filesystem $filesystem;
+    private LtiInstanceLoadBalancerFactory $ltiInstanceLoadBalancerFactory;
+    private BulkUserCreationCsvWriter $csvWriter;
+
+    private const DEFAULT_USERNAME_INCREMENT_VALUE = 0;
+
+    public function __construct(
+        LineItemRepository $lineItemRepository,
+        AssignmentRepository $assignmentRepository,
+        LtiInstanceLoadBalancerFactory $ltiInstanceLoadBalancerFactory,
+        BulkUserCreationCsvWriter $csvWriter,
+        Filesystem $filesystem,
+        string $generatedUsersFilePath
+    ) {
+        $this->lineItemRepository = $lineItemRepository;
+        $this->assignmentRepository = $assignmentRepository;
+        $this->ltiInstanceLoadBalancerFactory = $ltiInstanceLoadBalancerFactory;
+        $this->csvWriter = $csvWriter;
+        $this->generatedUsersFilePath = $generatedUsersFilePath;
+        $this->filesystem = $filesystem;
+    }
+
+    /**
+     * @throws LineItemNotFoundException
+     */
+    public function createUsers(
+        array $lineItemIds,
+        array $lineItemSlugs,
+        array $userPrefixes,
+        int $batchSize,
+        ?string $groupPrefix
+    ): UserCreationResult {
+
+        $notExistLineItemsArray = $userGroupIds = [];
+
+        if (!empty($lineItemIds) || !empty($lineItemSlugs)) {
+            $this->lineItemIds = $lineItemIds;
+            $this->lineItemSlugs = $lineItemSlugs;
+            $criteria = $this->getFindLineItemCriteria();
+            $lineItems = $this->lineItemRepository->findLineItemsByCriteria($criteria)->jsonSerialize();
+            if (empty($lineItems)) {
+                $exceptionMessage = $this->lineItemIds
+                    ? implode(',', $this->lineItemIds) . ' Line item id(s) not exist in the system'
+                    : implode(',', $this->lineItemSlugs) . ' Line item slug(s) not exist in the system';
+                throw new LineItemNotFoundException($exceptionMessage);
+            }
+
+            $this->generateSlugData($lineItems);
+            $notExistLineItemsArray = $this->lineItemIds
+                ? array_diff($lineItemIds, array_keys($this->lineItemSlugs))
+                : array_diff($lineItemSlugs, array_values($this->lineItemSlugs));
+        }
+
+        if (empty($lineItemIds) && empty($lineItemSlugs)) {
+            $this->getAllLineItemSlugs();
+        }
+
+        $userNameLastIndexes = $this->getLastUserAssignedToLineItems($this->lineItemSlugs);
+
+        $userGroupAssignCount = 0;
+        if ($groupPrefix) {
+            $loadBalancer = $this->ltiInstanceLoadBalancerFactory->__invoke('userGroupId');
```

Comment:
```suggestion
            $loadBalancer = $this->ltiInstanceLoadBalancerFactory('userGroupId');
```

Reply:
If i make the suggessted change getting some error : Call to undefined method OAT\SimpleRoster\Service\Bulk\BulkCreateUsersService::ltiInstanceLoadBalancerFactory()